### PR TITLE
Adding references to the documentation

### PR DESCRIPTION
**Pull request by raduom**
Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `$(nix-build default.nix -A pkgsLocal.updateMaterialized)` to update the materialized Nix files
   - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
   - [ ] `$(nix-shell shell.nix --run fix-stylish-haskell)` to fix any formatting issues
- If you changed any Purescript files:
   - [ ] `$(nix-shell shell.nix --run fix-purty)` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -37,4 +41,11 @@ You can also use a normal Haskell development environment on your computer to wr
    :caption: Plutus Reference
    :maxdepth: 2
 
+References
+----------
+
+1. We develop our software in Haskell_. If you need to gain a better understanding of this powerful programming language, a good place to start is the http://haskell.org site.
```
Comment (j-mueller) [inline doc/index.rst:47]:
```suggestion
1. Haskell_ is the programming language for Plutus contracts. 
```

**Review by j-mueller — APPROVED** [review]

File: doc/index.rst (hunk 1 above)
Comment (michaelpj) [inline doc/index.rst:47]: I don't *disagree* with these, but I guess I'm not sure how worth it is to *incrementally* improve these docs: most of these pages need to be "written properly", and if we do that we'll probably have to redo this too.

Comment (raduom) [reply doc/index.rst]: I think references wouldn't change that much, though.

Comment (j-mueller) [reply doc/index.rst]: There is a JIRA issue for "Playground references", that's what the PR is about

File: doc/index.rst [hunk 2]
```diff
@@ -37,4 +41,11 @@ You can also use a normal Haskell development environment on your computer to wr
    :caption: Plutus Reference
    :maxdepth: 2
 
+References
```
[the remark this answers is not in the document]
Comment (raduom) [reply doc/index.rst:44]: Doesn't the indentation finish the `toctree` definition?

If what you're saying is that it should be moved before the `toctree`, I don't find any problem with moving it, but it seemed like the `toctree` is more important than the `References`.

[the remark this answers is not in the document]
Comment (raduom) [reply doc/index.rst:44]: Nevermind, I see what you mean.

**Review by michaelpj — APPROVED** [review]

**Pull request merged**